components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -298,6 +301,31 @@ public static Map<String, String> getPlaceholderValues(EmailTemplate emailTempla
                             placeHolderData.put(placeHolder, "");
                         }
                     }
+                } else if (placeHolder.equals(NotificationConstants.EmailNotification.UTM_PARAMETERS_PLACEHOLDER)) {
+                    // Generate a single query param string with all UTM parameters
+                    StringBuilder utmParamStringBuilder = new StringBuilder();
+                    for (Map.Entry<String, String> entry : placeHolderData.entrySet()) {
+                        if (!entry.getKey().startsWith(
+                                NotificationConstants.EmailNotification.UTM_PARAMETER_PREFIX)) {
+                            continue;
+                        }
+                        if (utmParamStringBuilder.length() > 0) {
+                            utmParamStringBuilder.append("&");
```

Comment:
remove & at the end of query string

Reply:
there won't be a & at the end of the string. However the logic here could be improved. This is done with https://github.com/wso2-extensions/identity-event-handler-notification/pull/323/commits/48a6bf7e6e23466dcc95cb68cffa18e05ad5a54e.